api/src/controllers/listen.js

```
@@ -5,94 +5,88 @@ import config from '../config';
 import logger from '../utils/logger';
 import events from '../utils/events';
 
-exports.list = (req, res) => {
-	Listen.apiQuery(req.query)
-		.then(listens => {
-			res.json(listens);
-		})
-		.catch(err => {
-			logger.error(err);
-			res.status(422).send(err.errors);
-		});
+exports.list = async (req, res, _) => {
+	if (req.query.user && req.query.user != req.User.id) {
+		return res.sendStatus(403);
+	}
+
+	try {
+		const listens = await Listen.apiQuery(req.query);
+		res.json(listens);
+	} catch(err) {
+		logger.error({err});
+		res.status(422).send(err.errors);
+	}
 };
 
-exports.get = (req, res) => {
+exports.get = async (req, res, _) => {
 	if (req.params.listenId == 'undefined') {
 		return res.sendStatus(404);
 	}
 
-	Listen.findById(req.params.listenId)
-		.then(listen => {
-			if (!listen) {
-				return res.sendStatus(404);
-			}
+	try {
+		const listen = await Listen.findById(req.params.listenId);
+		if (!listen) {
+			return res.sendStatus(404);
+		}
+		if (listen.user.id != req.User.id) {
+			return res.sendStatus(403);
+		}
 
-			res.json(listen);
-		})
-		.catch(err => {
-			logger.error(err);
-			res.status(422).send(err.errors);
-		});
+		res.json(listen);
+	} catch(err) {
+		logger.error({err});
+		res.status(422).send(err.errors);
```

drop try/catch and let it blow up